test/integration_test/execute_rule/nonregistered_rule.js

```diff
@@ -0,0 +1,168 @@
+
+contract('TokenHolder::executeRule', async (accounts) => {
+
+  let accountProvider,
+    tokenHolder,
+    wallet1,
+    eip20TokenMock,
+    transferRule,
+    tokenRules,
+    ephemeralPrivateKey1,
+    ephemeralKeyAddress1,
+    keyData,
+    totalBalance = 500,
+    transferRule2;
+
+  describe('ExecuteRule integration test', async () => {
+
+    it('Should fail when non-registered rule calls executerule', async () => {
+
+      accountProvider = new AccountsProvider(accounts);
+
+      ( {
+        tokenHolder,
+        wallet1,
+        eip20TokenMock,
+        transferRule,
+        tokenRules,
+      } = await ExecuteRuleUtils.setup(accountProvider));
+
+      await eip20TokenMock.setBalance(tokenHolder.address, totalBalance);
+
+      ephemeralPrivateKey1 = '0xa8225c01ceeaf01d7bc7c1b1b929037bd4050967c5730c0b854263121b8399f3';
+      ephemeralKeyAddress1 = '0x62502C4DF73935D0D10054b0Fb8cC036534C6fb0';
+
+      let currentBlockNumber = await web3.eth.getBlockNumber(),
+        expirationHeight = currentBlockNumber + 50,
+        spendingLimit = 200;
+
+      await tokenHolder.submitAuthorizeSession(
+        ephemeralKeyAddress1,
+        spendingLimit,
+        expirationHeight,
+        { from: wallet1 },
+      );
+
+      keyData = await tokenHolder.ephemeralKeys(
+        ephemeralKeyAddress1,
+      );
+
+      let currentNonce = keyData.nonce,
+        amountTransferred = 50;
+
+      let nextAvailableNonce = currentNonce.toNumber() + 1;
+      const to = accountProvider.get();
+
+      const transferFromExecutable = await ExecuteRuleUtils.generateTransferFromExecutable(
+        tokenHolder.address,
+        to,
+        new BN(amountTransferred),
+      );
+
+      // It is not registered in TokenRules.
+      transferRule2 = await ExecuteRuleUtils.transferRule(tokenRules.address);
+
+      const { rsv } = await ExecuteRuleUtils.getExecuteRuleExTxData(
+        tokenHolder.address,
+        transferRule2.address,
+        transferFromExecutable,
+        new BN(nextAvailableNonce),
+        ephemeralPrivateKey1,
+      );
+
+      // 'to' address balance before calling executeRule method
+      assert.strictEqual(
+        (await eip20TokenMock.balanceOf(to)).cmp(new BN(0)),
+        0,
+      );
+
+      // tokenHolder balance before calling executeRule method
+      assert.strictEqual(
+        (await eip20TokenMock.balanceOf(tokenHolder.address)).cmp(
+          new BN(totalBalance)),
+        0,
+      );
+
+      const transactionResponse = await tokenHolder.executeRule(
+        transferRule2.address,
+        transferFromExecutable,
+        (currentNonce.toNumber() + 1),
+        rsv.v,
+        EthUtils.bufferToHex(rsv.r),
+        EthUtils.bufferToHex(rsv.s),
+      );
+
+      const events = Event.decodeTransactionResponse(
+        transactionResponse,
+      );
+
+      // We should check against false here, however current version of web3
+      // returns null for false values in event log. After updating web3,
```

Comment:
should we also add assert that nonRegisteredRule is not in TR registered rules mapping?

Reply:
Good Point.
Added an assert for it. Thank you.